### PR TITLE
MAINT: backports / preparation for SciPy 1.3.1

### PR DESCRIPTION
**Pull request by tylerjereddy**
Update release notes / version numbers & include backport PRs below. Please do check this over.

Backports:
https://github.com/scipy/scipy/pull/10431
https://github.com/scipy/scipy/pull/10379
https://github.com/scipy/scipy/pull/10309 (5 commits cherry-picked, unfortunately)
https://github.com/scipy/scipy/pull/10207
https://github.com/scipy/scipy/pull/10196
https://github.com/scipy/scipy/pull/10233 (this was directly merged to maintenance branch before the bump to "1.3.1 unreleased", unfortunately)

Could maybe add a short sentence summarizing the key bugfixes.

### CODE REVIEW

File: .mailmap [hunk 1]
```diff
@@ -73,6 +73,7 @@ Brett R. Murphy <bmurphy@enthought.com> brettrmurphy <bmurphy@enthought.com>
 Brian Hawthorne <brian.hawthorne@localhost> brian.hawthorne <brian.hawthorne@localhost>
 Brian Newsom <brian.newsom@colorado.edu> Brian Newsom <Brian.Newsom@Colorado.edu>
 Callum Jacob Hays <callumjhays@gmail.com> callumJHays <callumjhays@gmail.com>
+Carlos Ramos Carreno <vnmabus@gmail.com> vnmabus <vnmabus@gmail.com>
```
Comment (tylerjereddy) [inline .mailmap]: ping @vnmabus

Comment (vnmabus) [inline .mailmap]:
```suggestion
Carlos Ramos Carreño <vnmabus@gmail.com> vnmabus <vnmabus@gmail.com> 
```

File: doc/release/1.3.1-notes.rst [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+.. contents::
+
+SciPy 1.3.1 is a bug-fix release with no new features
+compared to 1.3.0.
+
+Authors
+=======
+
+* Peter Bell
+* Carlos Ramos Carreno +
```
Comment (vnmabus) [inline doc/release/1.3.1-notes.rst]:
```suggestion
* Carlos Ramos Carreño +
```

**Review by vnmabus — COMMENTED** [review]
Sorry for the correction, but in Spanish "n" and "ñ" are completely different characters.

**Comment by tylerjereddy** [comment]
I think I've identified the correct backport PR for the Travis issue, but the doc build issue is a little less clear. @mdhaber describes this in a comment here: https://github.com/scipy/scipy/pull/10286#issuecomment-502396266 Not sure if there's a single commit responsible for the eventual fix for that?

**Comment by pv** [comment]
Also #10503 would be nice.

**Comment by tylerjereddy** [comment]
I've updated this backport PR substantially based on recent fixes / backport-tagged PRs that have been merged.

In addition to the doc building fix I backported, I may also now need to grab another unrelated doc building fix from #10540, but I'll wait to confirm that this issue shows up here as well first.

There's an umerged PR with a backport label here: #10498

Details / new authors should be in the updated 1.3.1 release notes here.

**Comment by tylerjereddy** [comment]
Yeah, still a mess--I'll need more backports still. Some of the errors I do recognize from recent things at least, and perhaps the ones I don't are indirect failures from those..

**Comment by pv** [comment]
The qhull test fails since the old Qhull gives facets in different order.
I pushed a commit changing the test case to match the Qhull version in 1.3.0.

**Comment by tylerjereddy** [comment]
I've added several more backports, some of them from dissection of git history near points of failure in the source code with newer NumPy.

I'm hoping CI will be close to passing now. I still need to update the release notes with this latest batch of backports--I've kept some notes on what I've backported this evening so I can follow up with the release note changes a bit later.

**Comment by tylerjereddy** [comment]
ping @rainwoodman or @sturlamolden on the cdktree test failures in this branch

File: scipy/spatial/tests/test_kdtree.py [hunk 3]
```diff
@@ -1389,6 +1389,61 @@ def test_query_ball_point_length():
...
+    assert_array_equal(length1, length2)
+
+    d1, i1 = tree.query(query_contiguous, 1)
+    d2, i2 = tree.query(query_discontiguous, 1)
+
+    assert_array_equal(d1, d2)
+    assert_array_equal(i1, i2)
+
+
+@pytest.mark.parametrize("balanced_tree, compact_nodes",
```
Comment (sturlamolden) [inline scipy/spatial/tests/test_kdtree.py]: remove this test from the 1.3.x branch

**Comment by sturlamolden** [comment]
It looks like you got a test from #10457 (```test_cdktree_empty_input```) but not the rest of the commit. Either backport all of #10457 (it is just a few lines) or remove the test.

**Comment by tylerjereddy** [comment]
CI is *finally* passing -- I still need to update the release notes for the recent backport additions -- I'll likely do that this evening

**Comment by sturlamolden** [comment]
That is great :)

**Comment by tylerjereddy** [comment]
Pulled in another backport and updated the release notes for the batch of backports over last few days.

Let me know if any entries are missing / wrong -- I think I got them all.

**Comment by rgommers** [comment]
> Let me know if any entries are missing / wrong -- I think I got them all.

Indeed, didn't see any missing ones. 

Merged, thanks @tylerjereddy

**Comment by tylerjereddy** [comment]
Ok, I've created the source archives using cython 0.29.13 & wheels will build overnight--fingers crossed.

**Pull request merged**